internal/main.go

```diff
@@ -123,7 +137,7 @@ func getAuthsForImage(ctx context.Context, logger *slog.Logger, kr credentialpro
 	return auths
 }
 
-func pullImageWithRetries(ctx context.Context, logger *slog.Logger, wg *sync.WaitGroup, client criV1.ImageServiceClient, request *criV1.PullImageRequest, timing TimingConfig) {
+func pullImageWithRetries(ctx context.Context, logger *slog.Logger, wg *sync.WaitGroup, client criV1.ImageServiceClient, metricsSink chan<- *metricsProto.Result, name string, request *criV1.PullImageRequest, timing TimingConfig) {
```

Comment:
7 args?

Reply:
I failed when trying to come up with a way to considerably reduce this. I could make it 5 maybe with some aggregation into a struct, but it does not help much with the cognitive load, if not the contrary.

I'd say let's leave it for now as is, with 7 being the magic number of things that short term memory can fit. The types are sufficiently distinct not to cause issues due to incorrect order.